src/python/grpcio/tests/_runner.py

```diff
@@ -105,6 +105,8 @@ def write_bypass(self, value):
     Arguments:
       value (str): What to write to the original file.
     """
+    if sys.version_info.major >= 3 and not isinstance(value, bytes):
```

Comment:
`six.binary_type` would be equivalent (since `bytes` is an alias for `str` on python2), but I believe this check is to deal with bytearrays, not only to distinguish between binary and text data. On python3, bytearray literals were being printed instead of the contents of those bytearrays. So that makes me wonder whether using `six.binary_type` instead might be misleading?